cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/UpdateTopicConfigurationRunnable.java

```diff
@@ -48,38 +57,107 @@
+      _skipHardGoalCheck = topicReplicationFactorChangeParameters.skipHardGoalCheck();
+      _replicaMovementStrategy = topicReplicationFactorChangeParameters.replicaMovementStrategy();
+      _replicationThrottle = topicReplicationFactorChangeParameters.replicationThrottle();
+      _excludeRecentlyDemotedBrokers = topicReplicationFactorChangeParameters.excludeRecentlyDemotedBrokers();
+      _excludeRecentlyRemovedBrokers = topicReplicationFactorChangeParameters.excludeRecentlyRemovedBrokers();
+      _dryRun = topicReplicationFactorChangeParameters.dryRun();
+      _reason = topicReplicationFactorChangeParameters.reason();
+      _stopOngoingExecution = topicReplicationFactorChangeParameters.stopOngoingExecution();
+      }
     _uuid = uuid;
+    _isTriggeredByUserRequest = true;
   }
 
+  /**
+   * Constructor to be used for creating a runnable for self-healing.
+   */
+  public UpdateTopicConfigurationRunnable(KafkaCruiseControl kafkaCruiseControl,
+                                          Map<Short, Pattern> topicPatternByReplicationFactor,
+                                          List<String> selfHealingGoals,
+                                          boolean allowCapacityEstimation,
+                                          boolean excludeRecentlyDemotedBrokers,
+                                          boolean excludeRecentlyRemovedBrokers,
+                                          String anomalyId,
+                                          String reason) {
+    super(kafkaCruiseControl, new OperationFuture("Topic replication factor anomaly self-healing."));
+    _topicPatternByReplicationFactor = topicPatternByReplicationFactor;
+    _goals = selfHealingGoals;
+    _skipRackAwarenessCheck = SELF_HEALING_SKIP_RACK_AWARENESS_CHECK;
+    _requirements = SELF_HEALING_MODEL_COMPLETENESS_REQUIREMENTS;
+    _allowCapacityEstimation = allowCapacityEstimation;
+    _concurrentInterBrokerPartitionMovements = SELF_HEALING_CONCURRENT_MOVEMENTS;
+    _concurrentLeaderMovements = SELF_HEALING_CONCURRENT_MOVEMENTS;
+    _executionProgressCheckIntervalMs = SELF_HEALING_EXECUTION_PROGRESS_CHECK_INTERVAL_MS;
+    _skipHardGoalCheck = SELF_HEALING_SKIP_HARD_GOAL_CHECK;
+    _replicaMovementStrategy = SELF_HEALING_REPLICA_MOVEMENT_STRATEGY;
+    _replicationThrottle = kafkaCruiseControl.config().getLong(ExecutorConfig.DEFAULT_REPLICATION_THROTTLE_CONFIG);
+    _excludeRecentlyDemotedBrokers = excludeRecentlyDemotedBrokers;
+    _excludeRecentlyRemovedBrokers = excludeRecentlyRemovedBrokers;
+    _dryRun = SELF_HEALING_DRYRUN;
+    _reason = reason;
+    _stopOngoingExecution = SELF_HEALING_STOP_ONGOING_EXECUTION;
+    _uuid = anomalyId;
+    _isTriggeredByUserRequest = false;
+  }
+
+
   @Override
-  protected OptimizationResult getResult() throws Exception {
-    if (_topicReplicationFactorChangeParameters != null) {
+  public OptimizationResult getResult() throws Exception {
+    if (_topicPatternByReplicationFactor != null) {
```

Comment:
Null check on a random local variable of this runnable is a little confusing. Can we make it more clear?

Reply:
I feel this is needed because we plan to add more functionality to this Runnable.  It is supposed to be part of all the topic configuration change code path. Currently we only have topic RF change, so it looks a little bizarre. Once we support more topic configuration change type, it will make more sense.